apps/wallet/src/background/keyring/Account.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { Keypair, SuiAddress } from '@mysten/sui.js';
+
+export type AccountType = 'derived' | 'imported';
+
+export class Account {
+    #keypair: Keypair;
+    public readonly derivationPath: string | null;
+    public readonly address: SuiAddress;
+
+    constructor(
+        type: 'derived',
+        options: { derivationPath: string; keypair: Keypair }
+    );
+    constructor(type: 'imported', options: { keypair: Keypair });
+    constructor(
+        public readonly type: AccountType,
```

Comment:
Let's avoid using constructor property initializers and just assign it in the constructor body.